subsys/bluetooth/audio/unicast_client.c

```diff
@@ -0,0 +1,1956 @@
+	       pref->unframed_supported, pref->phy, pref->rtn, pref->latency,
+	       pref->pd_min, pref->pd_max, stream->codec->id);
+
+	unicast_client_ep_set_codec(ep, cfg->codec.id,
+				    sys_le16_to_cpu(cfg->codec.cid),
+				    sys_le16_to_cpu(cfg->codec.vid),
+				    buf, cfg->cc_len, NULL);
+
+	/* Notify upper layer */
+	if (stream->ops != NULL && stream->ops->configured != NULL) {
+		stream->ops->configured(stream, pref);
+	}
+}
+
+static void unicast_client_ep_qos_state(struct bt_audio_ep *ep,
+					struct net_buf_simple *buf)
+{
+	struct bt_ascs_ase_status_qos *qos;
+	struct bt_audio_stream *stream;
+
+	if (buf->len < sizeof(*qos)) {
+		BT_ERR("QoS status too short");
+		return;
+	}
+
+	stream = ep->stream;
+	if (stream == NULL) {
+		BT_ERR("No stream active for endpoint");
+		return;
+	}
+
+	qos = net_buf_simple_pull_mem(buf, sizeof(*qos));
+
+	ep->cig_id = qos->cig_id;
+	ep->cis_id = qos->cis_id;
+	(void)memcpy(&stream->qos->interval, sys_le24_to_cpu(qos->interval),
+		     sizeof(qos->interval));
+	stream->qos->framing = qos->framing;
+	stream->qos->phy = qos->phy;
+	stream->qos->sdu = sys_le16_to_cpu(qos->sdu);
+	stream->qos->rtn = qos->rtn;
+	stream->qos->latency = sys_le16_to_cpu(qos->latency);
+	(void)memcpy(&stream->qos->pd, sys_le24_to_cpu(qos->pd),
+		     sizeof(qos->pd));
+
+	BT_DBG("dir 0x%02x cig 0x%02x cis 0x%02x codec 0x%02x interval %u "
+	       "framing 0x%02x phy 0x%02x rtn %u latency %u pd %u",
+	       unicast_client_ep_is_snk(ep) ? BT_AUDIO_SINK : BT_AUDIO_SOURCE,
+	       ep->cig_id, ep->cis_id, stream->codec->id,
+	       stream->qos->interval, stream->qos->framing,
+	       stream->qos->phy, stream->qos->rtn, stream->qos->latency,
+	       stream->qos->pd);
+
+	/* Disconnect ISO if connected */
+	if (stream->iso->state == BT_ISO_CONNECTED) {
+		bt_audio_stream_disconnect(stream);
+	}
+
+	/* Notify upper layer */
+	if (stream->ops != NULL && stream->ops->qos_set != NULL) {
+		stream->ops->qos_set(stream);
+	}
+}
+
+static void unicast_client_ep_enabling_state(struct bt_audio_ep *ep,
+					     struct net_buf_simple *buf)
+{
+	struct bt_ascs_ase_status_enable *enable;
+	struct bt_audio_stream *stream;
+
+	if (buf->len < sizeof(*enable)) {
+		BT_ERR("Enabling status too short");
+		return;
+	}
+
+	stream = ep->stream;
+	if (stream == NULL) {
+		BT_ERR("No stream active for endpoint");
+		return;
+	}
+
+	enable = net_buf_simple_pull_mem(buf, sizeof(*enable));
+
+	BT_DBG("dir 0x%02x cig 0x%02x cis 0x%02x",
+	       unicast_client_ep_is_snk(ep) ? BT_AUDIO_SINK : BT_AUDIO_SOURCE,
+	       ep->cig_id, ep->cis_id);
+
+	unicast_client_ep_set_metadata(ep, buf, enable->metadata_len, NULL);
+
+	/* Notify upper layer */
+	if (stream->ops != NULL && stream->ops->enabled != NULL) {
+		stream->ops->enabled(stream);
+	}
+}
+
+static void unicast_client_ep_streaming_state(struct bt_audio_ep *ep,
+					      struct net_buf_simple *buf)
+{
+	struct bt_ascs_ase_status_enable *enable;
```

Comment:
Agreed. Added new structs for the streaming and disabling states. 

The variable's purpose is just to extract memory yes, as well as providing a shorthand way of verifying length. It could be removed and replaced by `sizeof(struct bt_ascs_ase_status_enable)`.